File: src/main/java/hiiragi283/material/api/capability/IOControllable.java
package hiiragi283.material.api.capability;

import org.jetbrains.annotations.NotNull;

public interface IOControllable {

    @NotNull
    Type getIOType();

    enum Type {

        INPUT(true, false),
        OUTPUT(false, true),
        GENERAL(true, true),
        CATALYST(false, false);

        public final boolean canInsert, canExtract;

        Type(boolean canInsert, boolean canExtract) {
            this.canInsert = canInsert;
            this.canExtract = canExtract;
        }

    }

}

File: src/main/kotlin/hiiragi283/material/api/capability/IOControllable.kt
package hiiragi283.material.api.capability

interface IOControllable {

    var ioType: Type

    enum class Type(val canInsert: Boolean, val canExtract: Boolean) {
        INPUT(true, false),
        OUTPUT(false, true),
        GENERAL(true, true),
        CATALYST(false, false)
    }

}